From 626000c22a92ad82f2030dd494d6218ae2b59d90 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:57 -0400
Subject: [ethernet] i40e: move nway reset

Message-id: <1409658042-22689-237-git-send-email-sassmann@redhat.com>
Patchwork-id: 90423
O-Subject: [RHEL7.1 PATCH 236/281] i40e: move nway reset
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

Just move nway reset up, will be used in the next patch.

Change-ID: Ice3b631fa2044debc5c4541b42872a48163f8452
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit a65997215be9f54dbc927b05fc8eb2fe55912d11)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
index 5b5e4e9..4ae10c9 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -412,6 +412,25 @@ no_valid_phy_type:
  return 0;
 }
 
+static int i40e_nway_reset(struct net_device *netdev)
+{
+ /* restart autonegotiation */
+ struct i40e_netdev_priv *np = netdev_priv(netdev);
+ struct i40e_pf *pf = np->vsi->back;
+ struct i40e_hw *hw = &pf->hw;
+ bool link_up = hw->phy.link_info.link_info & I40E_AQ_LINK_UP;
+ i40e_status ret = 0;
+
+ ret = i40e_aq_set_link_restart_an(hw, link_up, NULL);
+ if (ret) {
+  netdev_info(netdev, "link restart failed, aq_err=%d\n",
+       pf->hw.aq.asq_last_status);
+  return -EIO;
+ }
+
+ return 0;
+}
+
 /**
  * i40e_get_pauseparam -  Get Flow Control status
  * Return tx/rx-pause status
@@ -1125,25 +1144,6 @@ static int i40e_set_wol(struct net_device *netdev, struct ethtool_wolinfo *wol)
  return 0;
 }
 
-static int i40e_nway_reset(struct net_device *netdev)
-{
- /* restart autonegotiation */
- struct i40e_netdev_priv *np = netdev_priv(netdev);
- struct i40e_pf *pf = np->vsi->back;
- struct i40e_hw *hw = &pf->hw;
- bool link_up = hw->phy.link_info.link_info & I40E_AQ_LINK_UP;
- i40e_status ret = 0;
-
- ret = i40e_aq_set_link_restart_an(hw, link_up, NULL);
- if (ret) {
-  netdev_info(netdev, "link restart failed, aq_err=%d\n",
-       pf->hw.aq.asq_last_status);
-  return -EIO;
- }
-
- return 0;
-}
-
 static int i40e_set_phys_id(struct net_device *netdev,
        enum ethtool_phys_id_state state)
 {
-- 
1.7.1